gems/pending/spec/util/miq-hash_struct_spec.rb

```diff
@@ -122,4 +122,30 @@
       expect(hs.to_hash[:abc]).to eq(123)
     end
   end
+
+  context "Marshal" do
+    it "dump" do
+      expect(Marshal.dump(described_class.new)).to eq("\x04\bo:\x12MiqHashStruct\a:\n@hash{\x00:\x0E@key_typec\vSymbol")
+    end
+
+    it "load" do
+      new_obj = Marshal.load("\x04\bo:\x12MiqHashStruct\a:\n@hash{\x00:\x0E@key_typec\vSymbol")
+      expect(new_obj).to           be_kind_of(MiqHashStruct)
+      expect(new_obj._hash).to     be_kind_of(Hash)
+      expect(new_obj._key_type).to eq(Symbol)
+    end
```

Comment:
I don't feel strongly about it in this particular case, but as a general principle, I'd rather stick to the behaviour that we "actually" care about: that we can dump&load an instance, and it'll still work afterwards.

The specific string that it dumps to (and maybe also the `_hash`/`_key_type` methods?) feels more like implementation detail: if `@key_type` got dumped before `@hash`, I don't think we'd consider that a bug.

On the other hand, maybe we really do want to know that we can unmarshal that particular string, which may have been stored somewhere.